### PR TITLE
Let axis=-1 by default in take_along_axis to match NumPy.

### PR DESCRIPTION
**Pull request by carlosgmartin**
Let `axis=-1` by default in [`take_along_axis`](https://docs.jax.dev/en/latest/_autosummary/jax.numpy.take_along_axis.html) to match [NumPy](https://numpy.org/doc/stable/reference/generated/numpy.take_along_axis.html).

This change happened in NumPy 2.3.0, as stated [here](https://numpy.org/doc/stable/release/2.3.0-notes.html):

> The parameter `axis` in `numpy.take_along_axis` function has now a default value of `-1`. ([gh-28615](https://github.com/numpy/numpy/pull/28615))

### CODE REVIEW

**Comment by jakevdp** [comment]
Looks good - could you add a test? Just a one-off test somewhere around [here](https://github.com/jax-ml/jax/blob/49f3ff73cf00c4de617f064c6cb784babb658d0c/tests/lax_numpy_test.py#L4812) that asserts that not passing `axis` returns the same results as passing `axis=-1` explicitly. Thanks!

**Comment by carlosgmartin** [comment]
@jakevdp Done.

**Review by jakevdp — APPROVED** [review]
Thanks!
